app/Services/UpdatePasswordService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brewmap\Services;
+
+use Brewmap\Exceptions\User\OldPasswordMismatchException;
+use Illuminate\Contracts\Hashing\Hasher;
+
+class UpdatePasswordService
+{
+    protected Hasher $hash;
+
+    public function __construct(Hasher $hash)
+    {
+        $this->hash = $hash;
+    }
+
+    /**
+     * @throws OldPasswordMismatchException
+     */
+    public function updatePassword(array $input): void
+    {
+        if (!$this->hash->check($input["old_password"], auth()->user()->password)) {
+            throw new OldPasswordMismatchException();
+        }
+        auth()->user()->update([
+            "password" => $this->hash->make($input["password"]),
+        ]);
```

Comment:
You can try to inject `Illuminate\Contracts\Auth\Guard` the same way like you're doing with `Illuminate\Contracts\Hashing\Hasher`.

Then you could try to call `user()` on `guard` object instead using `auth()` helper function
```suggestion
        $this->guard->user()->update([
            "password" => $this->hash->make($input["password"]),
        ]);
```